registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/ABISHandlerUtil.java

```diff
@@ -1,222 +1,225 @@
+	@Autowired
+	Utilities utilities;
+
+	/** The packet info manager. */
+	@Autowired
+	private PacketInfoManager<Identity, ApplicantInfoDto> packetInfoManager;
+
+	/** The packet info dao. */
+	@Autowired
+	private PacketInfoDao packetInfoDao;
+
+	@Autowired
+	private IdRepoService idRepoService;
+
+	/**
+	 * Gets the unique reg ids.
+	 *
+	 * @param registrationId   the registration id
+	 * @param registrationType the registration type
+	 * @return the unique reg ids
+	 * @throws ApisResourceAccessException           the apis resource access
+	 *                                               exception
+	 * @throws IOException                           Signals that an I/O exception
+	 *                                               has occurred.
+	 * @throws                                       io.mosip.kernel.core.exception.IOException
+	 */
+	public List<String> getUniqueRegIds(String registrationId, String registrationType, ProviderStageName stageName) throws ApisResourceAccessException, JsonProcessingException, PacketManagerException, IOException {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
+				registrationId, "ABISHandlerUtil::getUniqueRegIds()::entry");
+		
+		String latestTransactionId = utilities.getLatestTransactionId(registrationId);
+
+		List<String> regBioRefIds = packetInfoDao.getAbisRefMatchedRefIdByRid(registrationId);
+
+		List<String> machedRefIds = new ArrayList<>();
+		List<String> uniqueRIDs = new ArrayList<>();
+		List<AbisResponseDetDto> abisResponseDetDtoList = new ArrayList<>();
+
+		if (!regBioRefIds.isEmpty()) {
+			List<AbisResponseDto> abisResponseDtoList = packetInfoManager.getAbisResponseRecords(regBioRefIds.get(0),
+					latestTransactionId, AbisConstant.IDENTIFY);
+			for (AbisResponseDto abisResponseDto : abisResponseDtoList) {
+				abisResponseDetDtoList.addAll(packetInfoManager.getAbisResponseDetails(abisResponseDto.getId()));
+			}
+			if (!abisResponseDetDtoList.isEmpty()) {
+				for (AbisResponseDetDto abisResponseDetDto : abisResponseDetDtoList) {
+					machedRefIds.add(abisResponseDetDto.getMatchedBioRefId());
+				}
+				if (!CollectionUtils.isEmpty(machedRefIds)) {
+					List<String> matchedRegIds = packetInfoDao.getAbisRefRegIdsByMatchedRefIds(machedRefIds);
+					if (!CollectionUtils.isEmpty(matchedRegIds)) {
+						List<String> processingRegIds = packetInfoDao.getWithoutStatusCodes(matchedRegIds,
+								RegistrationTransactionStatusCode.REJECTED.toString(), RegistrationTransactionStatusCode.PROCESSED.toString());
+						List<String> statusList=new ArrayList<>();
+						statusList.add(RegistrationTransactionStatusCode.PROCESSED.toString());
+						statusList.add(RegistrationTransactionStatusCode.PROCESSING.toString());
+						List<String> processedRegIds = packetInfoDao.getProcessedOrProcessingRegIds(matchedRegIds,
+								statusList);
+						uniqueRIDs = getUniqueRegIds(processedRegIds, registrationId, registrationType, stageName);

```

Comment:
As per the latest change, we may get the rids with status_code as PROCESSING in both the processingRegIds and processedRegIds, will this create any problem ?

Reply:
As per the code  processing records present in processedRegIds   will get again filtered in 106 line number method  by checking uin is available in id-repo and only processed rids will get  added to uniqueRIDs .In this case uniqueRIDs  line number 106 contains processed rids then below processing rids will add.It will not create issue.